src/main/scala/de/htwg/se/sudoku/model/fileIoComponent/fileIoMicroImpl/FileIO.scala

```diff
@@ -0,0 +1,100 @@
+package de.htwg.se.sudoku.model.fileIoComponent.fileIoMicroImpl
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpMethods.POST
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.ActorMaterializer
+import com.google.inject.Guice
+import com.google.inject.name.Names
+import de.htwg.se.sudoku.MicroSudokuModule
+import de.htwg.se.sudoku.model.fileIoComponent.FileIOInterface
+import de.htwg.se.sudoku.model.gridComponent.GridInterface
+import net.codingwell.scalaguice.InjectorExtensions.ScalaInjector
+import play.api.libs.json.Json
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContextExecutor, Future}
+import scala.util.{Success, Try}
+
+class FileIO extends FileIOInterface {
+  implicit val system: ActorSystem = ActorSystem()
+  implicit val materializer: ActorMaterializer = ActorMaterializer()
+  implicit val executionContext: ExecutionContextExecutor = system.dispatcher
+
+  override def load: Try[Option[GridInterface]] = {
+    val responseFuture = Http()
+      .singleRequest(HttpRequest(uri = "http://localhost:8089/sudoku/model/fileio/v1/load"))
```

Comment:
Is there an easy way to read this from the config? If not, hardcoding the URI should be okay for demonstration purposes (but students looking at the code will have an easier time if its done right ;))

Reply:
Yes, that has bothered me already but just came up with a solution: We can inject the host part including the port with Juice from the module and just add the rest of the part at this point.

`http://localhost:8089/` will be injected and `sudoku/model/fileio/v1/load` added in this request.
The rest of the path is constant and if the host part or the port changes, only the module has to be adapted.

I think this is quiet clean and nice...